libraries/bsg_manycore_cuda.cpp

```diff
@@ -2053,9 +2053,20 @@ static int hb_mc_device_tiles_set_config_symbols (hb_mc_device_t *device,
 
 
 
-		// Set tile's tile group index  __bsg_tile_group_id_x/y symbol.
+		// Set tile's tile group index  __bsg_tile_group_id_x/y symbols.
+		// and tile's flat tile group id (tg_id_y * grid_dim_x + tg_id_x)
+		// __bsg_tile_group_id_x/y symbols define the X/Y coordinates of
+		// a tile's tile group within it's father grid.
```

Comment:
I'd say "with respect to the tile group origin"

Reply:
Oh no the origin tile has nothing to do with the tile_group_id, which goes to show how badly I've explained this in the comment.
The grid is actually just a 2D array of tile groups that represent an application, and the __bsg_tile_group_id_x/y define the index of the tile group, they are just there to identify the tile group and distinguish it from other tile groups within that grid. 
The flat __bsg_tile_group_id is just a flattened representation of the __bsg_tile_group_id_x/y, or like a 1D index of that tile group inside the grid. Maybe I should write it like this in the comment, what do you think?

Reply:
I just find 'father grid' confusing. Relative to the grid origin?

Reply:
OK, that sounds good, how does it look now?